matscipy/dislocation.py

```diff
@@ -2827,148 +2702,251 @@ def displacements(self, bulk_positions, center,
-                                                symbol=symbol)
-        left_shockley.set_burgers(burgers_left)
-        # another Shockley partial
-        # burgers_right = alat * np.array([1, -2, 1.]) / 6. - default value
-        right_shockley = FCCScrewShockleyPartial(alat, C11, C12, C44,
-                                                 symbol=symbol)
-        super().__init__(left_shockley, right_shockley, burgers)
+class FCCScrewShockleyPartial(CubicCrystalDislocation):
+    crystalstructure="fcc"
+    axes = np.array([[1, 1, -2],
+                     [1, 1, 1],
+                     [1, -1, 0]])
+    burgers_dimensionless = np.array([1, -2, 1.]) / 6.
+    glide_distance_dimensionless = np.sqrt(6) / 4.0
+    n_planes = 2
+    unit_cell_core_position_dimensionless = np.array([5/6, 1/9, 0])
 
 
 class FCCEdgeShockleyPartial(CubicCrystalDislocation):
-    def __init__(self, alat, C11, C12, C44, symbol='Fe'):
-        axes = np.array([[1, -1, 0],
-                         [1, 1, 1],
-                         [-1, -1, 2]])
-
-        burgers = alat * np.array([1, -2, 1.]) / 6.
-
-        parity = [0, 0]
+    crystalstructure = "fcc"
+    axes = np.array([[1, -1, 0],
+                     [1, 1, 1],
+                     [-1, -1, 2]])
+    burgers_dimensionless = np.array([1, -2, 1.]) / 6
+    unit_cell_core_position_dimensionless = np.array([0, 1/6 , 0])
+    glide_distance_dimensionless = np.sqrt(2)/4
+    n_planes = 6
 
-        unit_cell = FaceCenteredCubic(symbol, directions=axes.tolist(),
-                                      pbc=(False, False, True),
-                                      latticeconstant=alat)
 
-        disloCenterX = 0.0
-        # middle between two (111) planes
-        disloCenterY = unit_cell.cell[1][1] / 6.0
-
-        unit_cell_core_position = np.array([disloCenterX,
-                                            disloCenterY, 0])
+class FCCScrew110Dislocation(CubicCrystalDissociatedDislocation):
+    burgers_dimensionless = np.array([1, -1, 0]) / 2
+    new_left_burgers = np.array([2., -1., -1.]) / 6
+    left_dislocation = FCCScrewShockleyPartial
+    right_dislocation = FCCScrewShockleyPartial
 
-        glide_distance = alat * np.linalg.norm(axes[0]) / 4.0
 
-        n_planes = 6
-        self_consistent = True
-        super().__init__(unit_cell, alat, C11, C12, C44,
-                         axes, burgers, unit_cell_core_position, parity,
-                         glide_distance, n_planes=n_planes,
-                         self_consistent=self_consistent)
+class FCCEdge110Dislocation(CubicCrystalDissociatedDislocation):
+    crystalstructure = "fcc"
+    burgers_dimensionless = np.array([1, -1, 0]) / 2
+    new_left_burgers = np.array([2., -1., -1.]) / 6
+    left_dislocation = FCCEdgeShockleyPartial
+    right_dislocation = FCCEdgeShockleyPartial
+
+
+def _validate_disloc_inputs(a, symbol="w", axes=None, crystalstructure="fcc", pbc=[True, True, True]):
```

Comment:
to be moved out of this module

Reply:
`pbc` is initialised to a mutable object. Should be set to `None` and then assigned within function.

Reply:
Changed pbc=True to be the default, which should retain the default behaviour without requiring extra variables.